src/devices/acurite.c

```diff
@@ -996,6 +996,13 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
                 valid++;
             }
             else if (message_type == ACURITE_MSGTYPE_RAINFALL) {
+                // If the row is missing zeros, it can still pass the checksum.
+                // The earlier length check passes as ACURITE_TXR_BITLEN is 56.
```

Comment:
I don't like this comment, but I admit up front I am nitpicking.

First, it's a truism that corrupted messages could still pass a checksum, and we don't check other things only because of that -- we check everything we can, meaning every property that is believed to be true of all messages.

Second, and this is the bigger thing, the "earlier length check" is confusing.  I see 56, and that feels like 7.   If the earlier check is "every valid message is >= 7 bytes", then it's great to reject early.   But if different types are different lengths, then every type should have a length check.   And if the longest messages is 12 bytes (totally making that up) then the earlier check should also require <= 12.

Finally, the next line says <8, but is 9 valid?  20?  Should this be ==8?   If not ==, is there some upper bound?  If it is variable, is there an embedded length that can be checked after the min check?

I realize my specific comments may be off here, but I hope the approach of trying to check as aggressively as possible is useful.

Reply:
Thanks for the feedback, it's appreciated!

> But if different types are different lengths, then every type should have a length check.

I agree. I think this function is somewhat confusing because it combines several different message formats all at once. And given that we're working off of reverse engineered samples (and not committed specifications), it's entirely possible that future devices emit valid signals that break these assumptions.

I'm tempted to consider refactoring this along with the return codes mentioned above, but would probably do that as a followup PR as it's hard for me to know exactly how much time I'll have to commit in the next few months.

> Finally, the next line says <8, but is 9 valid? 20? Should this be ==8?

I'm not sure how to confirm that. I did `<` because that seemed to be the most targeted fix for corrupted messages. I'm OK changing it to 8, and if a future user somehow finds a valid message that is longer, at least we can ask for a sample to be included in the tests.